app/controllers/application_controller.rb

```diff
@@ -1,10 +1,160 @@
 require './config/environment'
-
+require 'pry'
 class ApplicationController < Sinatra::Base
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+  
+  get '/' do 
+    erb :index 
+  end
+  
+  get '/signup' do 
+    if logged_in?
+      redirect "/tweets"
+    else
+      erb :"users/create_user"
+    end
+  end
+  
+  post '/signup' do 
+    if params["username"] == "" || params["email"] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.